anomalib/models/components/base/anomalib_module.py

```diff
@@ -0,0 +1,190 @@
+"""Base Anomalib Module for Train/Test/Inference Tasks."""
```

Comment:
Should we remove `AnomalyModule`? Or are you planning to do this in a later PR?

Reply:
yes, when we deprecate the existing lightning modules we'll deprecate anomaly_module as well